### ISSUE TITLE
Jumpy scrolling

### ISSUE THREAD

**Issue by tsujan** [lxde/pcmanfm-qt#579]
Scrolling by 3 items is OK as for the speed but it's jumpy in pacmanfm-qt (and file dialog), especially with icon sizes ≥ 48px. When the view is scrolled by the mouse wheel, sometimes it's hard to keep track of items because their positions suddenly change. I sometimes have to grab the scrollbar to know where I am inside a folder like `/usr/share/applications`.

I always had a subconscious feeling about the difference between Dolphin's and pcmanfm-qt's scrolling behaviors. Dolphin *animates* the scrolling, so that the user never loses track of items visually, although the scroll speed is the same as in pcmanfm-qt.

IMHO, an animation would be very practical in this case.

**Comment by tsujan** [comment]
Added an animation -- without uploading -- but the CPU usage wan't acceptable (although not high)... should find and try other methods....

**Comment by tsujan** [comment]
Progress report:

The extra CPU usage is fixed by choosing lower frames per second with fast scrolling. But, although I chose a high FPS for slow scrolling, the animation isn't as smooth as it should be. There should be a hidden key somewhere ;)

**Comment by MggMuggins** [comment]
Seems like it's a little fast at this point. Is there any way to retain the performance but slow things down?

**Comment by tsujan** [comment]
> Is there any way to retain the performance but slow things down?

Yes, to some extent! I've set it to 300 ms in the patch I'm using/testing. More than that would mean waiting longer for the animation to end, which wouldn't be desirable after a while. As far as I've tested, it's enough to see items going up or down smoothly. Dolphin's animation is much faster.

The next step is testing with variable frames per second: higher for slow scrolling and lower for fast scrolling. This might not be possible if a speed of 3 rows per wheel turn (or whatever value that's set in the mouse settings dialog) should always be respected. Dolphin doesn't respect it but I prefer pcmanfm-qt to do so.
